velox/connectors/hive/HiveDataSink.h

```diff
@@ -209,15 +209,17 @@ class HiveInsertTableHandle : public ConnectorInsertTableHandle {
       // When this option is set the HiveDataSink will always write a file even
       // if there's no data. This is useful when the table is bucketed, but the
       // engine handles ensuring a 1 to 1 mapping from task to bucket.
-      const bool ensureFiles = false)
+      const bool ensureFiles = false,
+      const std::vector<std::string> partitionKeys = {})
       : inputColumns_(std::move(inputColumns)),
         locationHandle_(std::move(locationHandle)),
         storageFormat_(storageFormat),
         bucketProperty_(std::move(bucketProperty)),
         compressionKind_(compressionKind),
         serdeParameters_(serdeParameters),
         writerOptions_(writerOptions),
-        ensureFiles_(ensureFiles) {
+        ensureFiles_(ensureFiles),
+        partitionKeys_(partitionKeys) {
```

Comment:
Add a validation that if this is non-empty, we make sure every partition column appears here

Reply:
@Yuhta Added.